pkg/uploader/kopia/snapshot.go

```diff
@@ -80,6 +85,9 @@ func setupDefaultPolicy(ctx context.Context, rep repo.RepositoryWriter, sourceIn
 		SchedulingPolicy: policy.SchedulingPolicy{
 			Manual: true,
 		},
+		ErrorHandlingPolicy: policy.ErrorHandlingPolicy{
+			IgnoreUnknownTypes: newOptionalBool(true),
```

Comment:
Ignoring errors arbitrarily is dangerous, we cannot set it to `true` by default.
If we want to support to ignore unknown errors, the option should be disabled by default and we only enable it from a users' option.
Therefore, we have to create a user interface for this feature.

Finally, personally I think this option is not Must Have. As long as we can capture the error message and report the users(as for ErrorHandlingPolicy) it is good enough.

Reply:
In fact, the unknown type of files is not regular files at the filesystem level, including setuid/gid, hardlinks, mount points, and UNIX socket. both of them are not usable after restore even if Kopia could backup and users also won't expect these could work. so it's not dangerous if we could show the ignored files as a warning in the backup CR maybe that is enough.

Reply:
Checked Kopia code further, this IgnoreUnknownTypes option is checked for the file system entries that are not recognized by Kopia uploader. Kopia uploader will fail to back up these entries inevitably and IgnoreUnknownTypes is only to control whether the uploader error out or ignore and process.
However, it doesn't mean these entries are not useful, after all, we even don't know what they are exactly (Kopia at present supports dirs, files and symlinks only, other entry types all go to the UnknownTypes, a.k.a., ErryEntries). Instead, it only means Kopia uploader doesn't support them, but since we are using Kopia uploader, we don't have any way to support them if the uploader doesn't support.

For Kopia itself, it by default sets IgnoreUnknownTypes to true and gives users options to modify it. For Velero:

- If we set it to false, we will face some embarrassment that  Kopia util doesn't fail itself but Velero fails
- If we set it to true, as mentioned above, the unknown entries are still meaningful to users applications, we will also face the embarrassment that when users really fail the backup on these unknown entries, Kopia util can but Velero cannot

Therefore, the best solution is to follow Kopia -- set it to true by default and expose a user interface.

Reply:
yes, I set IgnoreUnknownTypes to `True` is also refers to the default sets of Kopia.

if we want users to set IgnoreUnknownTypes by themself, what about exposing the interface for users to set all Kopia [polices](https://github.com/kopia/kopia/blob/v0.10.4/snapshot/policy/policy.go#L23) in case of we adding options one by one in future ?